web/src/hotspots.ts

```diff
@@ -304,14 +343,17 @@ function close_read_hotspots(new_hotspots) {
     }
 }
 
-export function open_popover_if_hotspot_exist(hotspot_name, bind_element = null) {
-    const overlay_name = "hotspot_" + hotspot_name + "_overlay";
+export function open_popover_if_hotspot_exist(
+    hotspot_name: string,
+    bind_element: HTMLElement,
+): void {
+    const overlay_name: string = "hotspot_" + hotspot_name + "_overlay";
 
     if (is_hotspot_displayed(hotspot_name)) {
         overlays.open_overlay({
             name: overlay_name,
             $overlay: $(`#${CSS.escape(overlay_name)}`),
-            on_close: function () {
+            on_close: function (this: HTMLElement) {
```

Comment:
Does this work?

Reply:
Yes, here we are defining the type of "this"

Reply:
Though this code might just be better written using `bind_element` rather than the `.bind` nonsense, and not using `this` as all? Whatever, we're probably deleting this code in a few weeks anyway.